app/controllers/comments_controller.rb

```diff
@@ -28,6 +28,8 @@ def create
       # includes parent story_id to ensure this comment's story_id matches
       comment.parent_comment =
         Comment.find_by(story_id: story.id, short_id: params[:parent_comment_short_id])
+      comment.depth = comment.parent_comment.depth + 1
```

Comment:
I'd also rather call `update_column` (or a similar method) with the `.parents.count` to make the database count. I know this is very simple and correct now, but I do have some feature ideas around being able to move comments between parents.

Reply:
Discussed on stream. @rahul1990gupta noted that my comment is incorrect for this context - the comment is a new record that is not yet persisted to the database. So we can't call `update_column` or similar.

What we really want to say is that the value of depth is an SQL expression so that the database is always consistent, but ActiveRecord doesn't allow us to express that. A race condition is only theoretical, especially because I haven't actually written a feature that might change a parent's depth at the time someone is posting a comment.

I'll write a checker to watch for validity problems and we can address this when those come up. No change to this code.